utils/src/serialization.rs

```diff
@@ -100,7 +100,7 @@ where
         S: serde::Serializer,
     {
         let mut bytes = vec![];
-        val.serialize_unchecked(&mut bytes)
+        val.serialize_uncompressed(&mut bytes)
```

Comment:
Are we sure it doesn't check? It would imply a performance regression otherwise. I would be happy to see some regression tests for this.